x-pack/plugins/ml/public/application/explorer/alerts/swim_lane_wrapper.tsx

```diff
@@ -0,0 +1,273 @@
+import {
+  EuiButtonIcon,
+  EuiDescriptionList,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiInMemoryTable,
+  EuiNotificationBadge,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiText,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { ALERT_DURATION, ALERT_RULE_NAME, ALERT_START } from '@kbn/rule-data-utils';
+import { pick } from 'lodash';
+import React, { type FC, useMemo, useRef } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { i18n } from '@kbn/i18n';
+import { EuiBasicTableColumn } from '@elastic/eui/src/components/basic_table/basic_table';
+import { PanelHeaderItems } from '../../components/collapsible_panel';
+import { AnomalyDetectionAlert } from './anomaly_detection_alerts_state_service';
+import {
+  ALERT_ANOMALY_DETECTION_JOB_ID,
+  ALERT_ANOMALY_TIMESTAMP,
+  alertFieldNameMap,
+} from '../../../../common/constants/alerts';
+import {
+  getAlertEntryFormatter,
+  getAlertFormatters,
+} from '../../../alerting/anomaly_detection_alerts_table/render_cell_value';
+import { useMlKibana } from '../../contexts/kibana';
+import { useAnomalyExplorerContext } from '../anomaly_explorer_context';
+import type { AppStateSelectedCells, SwimlaneData } from '../explorer_utils';
+import { Y_AXIS_LABEL_WIDTH } from '../swimlane_annotation_container';
+import { CELL_HEIGHT } from '../swimlane_container';
+import { statusNameMap } from './const';
+
+export interface SwimLaneWrapperProps {
+  selection?: AppStateSelectedCells | null;
+  swimlaneContainerWidth?: number;
+  swimLaneData: SwimlaneData;
+}
+
+/**
+ * Wrapper component for the swim lane
+ * that handles the popover for the selected cells.
+ */
+export const SwimLaneWrapper: FC<SwimLaneWrapperProps> = ({
+  children,
+  selection,
+  swimlaneContainerWidth,
+  swimLaneData,
+}) => {
+  const {
+    services: { fieldFormats },
+  } = useMlKibana();
+
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const { anomalyDetectionAlertsStateService, anomalyTimelineStateService } =
+    useAnomalyExplorerContext();
+
+  const selectedAlerts = useObservable(anomalyDetectionAlertsStateService.selectedAlerts$, []);
+
+  const leftOffset = useMemo<number>(() => {
+    if (!selection || !swimLaneData) return 0;
+    const selectedCellIndex = swimLaneData.points.findIndex((v) => v.time === selection.times[0]);
+    const cellWidth = swimlaneContainerWidth! / swimLaneData.points.length;
+
+    const cellOffset = (selectedCellIndex + 1) * cellWidth;
+
+    return Y_AXIS_LABEL_WIDTH + cellOffset;
+  }, [selection, swimlaneContainerWidth, swimLaneData]);
+
+  const popoverOpen = !!selection && !!selectedAlerts?.length;
+
+  const button = (
+    <button
+      data-test-subj="mlSwimLanePopoverTrigger"
+      css={css`
+        position: absolute;
+        top: 0;
+        visibility: hidden;
+      `}
+    />
+  );
+
+  const alertFormatter = useMemo(() => getAlertEntryFormatter(fieldFormats), [fieldFormats]);
+
+  const viewType = 'table';
```

Comment:
Is there a different viewType other than table?

Reply:
There were another view in the beginning and then we came up with the table layout. I kept both for now to chose from or in case we use different view based on the number of results.